server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -6678,6 +6678,12 @@ public UserVm restoreVirtualMachine(final Account caller, final long vmId, final
                 newVol = volumeMgr.allocateDuplicateVolume(root, null);
             }
 
+            if (userVmDetailsDao.findDetail(vm.getId(), VmDetailConstants.ROOT_DISK_SIZE) == null && !newVol.getSize().equals(template.getSize())) {
+                VolumeVO resizedVolume = (VolumeVO) newVol;
+                resizedVolume.setSize(template.getSize());
+                _volsDao.update(resizedVolume.getId(), resizedVolume);
```

Comment:
@shwstppr it is possible to handle this case while allocating duplicate volume with template ?

Reply:
@sureshanaparti Duplicating using template, won't that affect other existing values of the entry - folder, path, etc?

Reply:
I missed this comment, if we need to address anything additional pl raise a new PR @shwstppr @sureshanaparti